docs/tutorials/projects.md

```diff
@@ -0,0 +1,347 @@
+
+<div class="terminal">
+```bash
+$ prefect deploy -f 'log-flow' \
+    -n my-docker-git-deployment \
+    -p docker-work
+$ prefect deployment run 'log-flow/my-docker-git-deployment'
+```
+</div>
+
+As promised above, this worked out of the box!  
+
+Let's deploy a new flow from this project that requires additional dependencies that might not be available in the default image our work pool is using; this flow requires both `pandas` and `numpy` as a dependency, which we will also install locally first and confirm the flow is working:
+
+<div class="terminal">
+```bash
+$ pip install -r requirements.txt
+$ python flows/pandas_flow.py
+```
+</div>
+
+We now have two options for how to manage these dependencies in our worker's environment:
+
+- setting the `EXTRA_PIP_PACKAGES` environment variable or using another hook to install the dependencies at runtime
+- building a custom Docker image with the dependencies baked in
+
+The first option is the quickest, but it does run the risk of changing runtime state from one run to the next and is not recommended for production use cases.  For the second option, we need to configure a `build` step within our `prefect.yaml` file as follows (Note: if starting from scratch we could use the `docker-git` recipe):
+
+```yaml
+# partial contents of prefect.yaml
+
+build:
+- prefect_docker.projects.steps.build_docker_image:
+    image_name: local-only/testing
+    tag: dev
+    dockerfile: auto
+    push: false
+
+pull:
+- prefect.projects.steps.git_clone_project:
+    repository: https://github.com/PrefectHQ/hello-projects.git
+    branch: main
+```
+
+A few notes:
+
+- [each step](/concepts/projects/#the-prefect-yaml-file) references a function with inputs and outputs
+- in this case, we are using `dockerfile: auto` to tell Prefect to automatically create a `Dockerfile` for us; otherwise we could write our own and pass its location as a path to the `dockerfile` kwarg
+- to avoid dealing with real image registries, we are not pushing this image; in most use cases you will want `push: true` (which is the default)
+
+All that's left to do is create our deployment and specify our image name to instruct the worker what image to pull:
+
+<div class="terminal">
+```bash
+$ prefect deploy -f 'pandas-flow' \
+    -n docker-build-deployment \
+    -p docker-work \
+    -v image=local-only/testing:dev
+$ prefect deployment run 'log-flow/my-docker-git-deployment'
+```
+</div>
+
+Your run should complete succsessfully, logs and all!  Note that the `-v` flag represents a job variable, which are the allowed pieces of infrastructure configuration on a given work pool.  Each work pool can customize the fields they accept here.
+
+!!! tip "Templating values"
+    As a matter of best practice, you should avoid hardcoding the image name and tag in both your `prefect.yaml` and CLI. Instead, you should [use variable templating](/concepts/projects/#templating-options).
+
+#### A counterexample
```

Comment:
Totally agree, though the title confused me. At first I thought it was retroactively arguing that the previous section wasn't correct for some reason. I think it could be more precise ("Converting local deployments to Docker") or something

Reply:
@jlowin you clearly don't read enough math

Reply:
I would call it "Failing early and explicitly with workers" which I think its the primary value prop you are explaining.